impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java

```diff
@@ -120,6 +135,17 @@ public static <T> T get(BeanManager beanManager, Type type, boolean create, Anno
         }
     }
 
+    private static String getBeanName(Bean<?> bean)
+    {
+        String name = bean.getName();
+        if (name != null)
+        {
+            return name;
+        }
+        String className = bean.getBeanClass().getSimpleName();
+        return Character.toLowerCase(className.charAt(0)) + className.substring(1);
```

Comment:
These two lines are the only ones I am not sure about but @tandraschko will weigh in!

Reply:
i also wonder about this one - why is this required?
If bean.getName is null, its not a named bean - why would need it then?

Reply:
TBH I'm not sure too - guess it depends on how the validator is registered in CDI?

What I'm sure is that without these lines the fix doesn't work...

Reply:
It looks like BalusC does this in his CDIUtils for good reason I guess.  he does...

```java
        String className = bean.getBeanClass().getSimpleName();

        return new StringBuilder()
            .append(Character.toLowerCase(className.charAt(0)))
            .append(className.substring(1))
            .toString();
```